src/org/opensolaris/opengrok/analysis/perl/PerlXref.lex

```diff
@@ -41,17 +43,27 @@ import org.opensolaris.opengrok.web.Util;
 %int
 %char
 %init{
-        h = new PerlLexHelper(QUO, QUOxN, QUOxL, QUOxLxN, this,
-            HERE, HERExN, HEREin, HEREinxN);
+    h = getNewHelper();
 %init}
 %{
+    private PerlLexHelper h;
+
   // TODO move this into an include file when bug #16053 is fixed
   @Override
   protected int getLineNumber() { return yyline; }
   @Override
   protected void setLineNumber(int x) { yyline = x; }
 
-    private final PerlLexHelper h;
+    /**
+     * Reinitialize the lexer with new reader.
+     *
+     * @param reader new reader for this lexer
+     */
+    @Override
+    public void reInit(Reader reader) {
+        super.reInit(reader);
+        h = getNewHelper();
```

Comment:
???
the design of analyzers should be that they are static and can statically process any new stream ...
if you create new object in every reinit it kinda beats this design ... are you sure you want to do this?
it also will break contract with lucene

Reply:
Oh I do not agree that they are static. There is a lot of state in a jflex class, and even more in the OpenGrok JFlexTokenizer and JFlexXref abstract classes.